bindings/wasm/examples/gltf-io.js

```diff
@@ -0,0 +1,208 @@
+
+import {Accessor} from '@gltf-transform/core';
+
+import {EXTManifold} from './manifold-gltf.js';
+
+export const attributeDefs = {
+  'POSITION': {type: Accessor.Type.VEC3, components: 3},
+  'SKIP': {type: null, components: 1},
+  'NORMAL': {type: Accessor.Type.VEC3, components: 3},
+  'TANGENT': {type: Accessor.Type.VEC4, components: 4},
+  'TEXCOORD_0': {type: Accessor.Type.VEC2, components: 2},
+  'TEXCOORD_1': {type: Accessor.Type.VEC2, components: 2},
+  'COLOR_0': {type: Accessor.Type.VEC3, components: 3},
+  'JOINTS_0': {type: Accessor.Type.VEC4, components: 4},
+  'WEIGHTS_0': {type: Accessor.Type.VEC4, components: 4},
+};
+
+export function setupIO(io) {
+  return io.registerExtensions([EXTManifold]);
+}
+
+export function readMesh(mesh, attributes, materials) {
+  if (attributes.length < 1 || attributes[0] !== 'POSITION')
+    throw new Error('First attribute must be "POSITION".');
+
+  const primitives = mesh.listPrimitives();
+  if (primitives.length === 0) {
+    return {};
+  }
+
+  const numProp = attributes.map((def) => attributeDefs[def].components)
+                      .reduce((a, b) => a + b);
+  const position = primitives[0].getAttribute('POSITION');
+  const numVert = position.getCount()
+  const vertProperties = new Float32Array(numProp * numVert);
+  let offset = 0;
+  for (const attribute of attributes) {
+    const accessor = primitives[0].getAttribute(attribute);
+    const array = accessor.getArray();
+    const size = accessor.getElementSize();
+    for (let i = 0; i < numVert; ++i) {
+      for (let j = 0; j < size; ++j) {
+        vertProperties[numProp * i + offset + j] = array[i * size + j];
+      }
+    }
+    offset += size;
+  }
+
+  const triVertArray = [];
+  const runIndexArray = [0];
+  for (const primitive of primitives) {
+    if (primitive.getAttribute('POSITION') === position) {
+      triVertArray.push(...primitive.getIndices().getArray());
+      runIndexArray.push(triVertArray.length);
+      materials.push(primitive.getMaterial());
+    } else {
+      console.log('primitives do not share accessors!');
+    }
+  }
+  const triVerts = new Uint32Array(triVertArray);
+  const runIndex = new Uint32Array(runIndexArray);
+
+  const manifoldPrimitive = mesh.getExtension('EXT_manifold');
+  const mergeTriVert =
+      manifoldPrimitive ? manifoldPrimitive.getMergeIndices().getArray() : [];
+  const mergeTo =
+      manifoldPrimitive ? manifoldPrimitive.getMergeValues().getArray() : [];
+  const vert2merge = new Map();
+  for (const [i, idx] of mergeTriVert.entries()) {
+    vert2merge.set(triVerts[idx], mergeTo[i]);
+  }
+  const mergeFromVert = [];
+  const mergeToVert = [];
+  for (const [from, to] of vert2merge.entries()) {
+    mergeFromVert.push(from);
+    mergeToVert.push(to);
+  }
+
+  return {
+    numProp,
+    triVerts,
+    vertProperties,
+    runIndex,
+    mergeFromVert,
+    mergeToVert
+  };
+}
+
+export function writeMesh(doc, manifoldMesh, attributes, materials) {
```

Comment:
This file is designed for interop specifically between Manifold and glTF. It doesn't actually have Manifold as a dependency because it can just use generic objects that look like a `Mesh`, but it's still based on our API.